### ISSUE TITLE
delete old log file with date being part of log file name

### ISSUE THREAD

**Issue by slav** [NLog/NLog#390]
From docs on File-Target in One log file per day section:
```
        <target name="file" xsi:type="File"
            layout="${longdate} ${logger} ${message}" 
            fileName="${basedir}/${shortdate}.log" />
```
This is exactly what I want to do. BUT, I want to only keep last 30 files, and delete file from 31 days ago.

I tried to test it with this setup
```
		<target name="fileAll" xsi:type="File" fileName="${basedir}/${date:format=yyyyMMdd-HHmm}.log"
		layout="${defaultLayout}" archiveEvery="minute" maxArchiveFiles="2" archiveFileName="${basedir}/{#}.log" archiveDateFormat="yyyyMMdd-HHmm" archiveNumbering="Date"/>
```

But files didn't get deleted. It's as if archive isn't working in this specific case. Using NLog 3.1.

Any advice on how to set it up to work?

The reason for this setup is because I want the logfile to be shipped. For log file shipped to have less problems it's ideal for the file to never be renamed after it's created. So, I don't want to have log.txt that later becomes log.{date}.txt. I want it to be log.{date}.txt from the beginning and only to be deleted in 30 days.

Thank you.

**Comment by rbouny** [comment]
+1

**Comment by testzcrypto** [comment]
+1

**Comment by 304NotModified** [comment]
moved to 4.2 as it's not clear how to fix this (we need to investigate further)

**Comment by 304NotModified** [comment]
We fixed some code related to this in 4.1. We should test after release.

**Comment by 304NotModified** [comment]
I'm working on this. Thanks for the +1. 

We are now using http://nlog.userecho.com/ :)

**Comment by 304NotModified** [comment]
I can confirm this bug has been fixed in 4.1. See unit tests in https://github.com/NLog/NLog/pull/986 (Tested in 4.1 and 4.0, and some relevant tests are failing in 4.0 and succeeding in 4.1)
